cmd/kube-apiserver/app/server.go

```diff
@@ -144,7 +148,11 @@ func Run(runOptions *options.ServerRunOptions, stopCh <-chan struct{}) error {
 	if err != nil {
 		return err
 	}
-	aggregatorServer, err := createAggregatorServer(aggregatorConfig, kubeAPIServer.GenericAPIServer, sharedInformers, apiExtensionsServer.Informers)
+	var aggregatorTranport *http.Transport
+	if tunneler != nil {
```

Comment:
I think this is better fixed with the "single thing". CreateDialer always creates a proxyTransport, even if its just a vanilla http.Transport, because CreateKubeAPIServerConfig sets it in master.Config. From there it gets passed around several different paths, including in to NewStorage. I wasn't sure it would be safe to eliminate but wanted to be clear whether we were using a SSHTunnel or not.